link_shorterner/src/shortener_shortener.erl

```diff
@@ -0,0 +1,53 @@
+-module(shortener_shortener).
+
+-export([short/1, get/1]).
+
+-export([start_link/0, init/1, handle_call/3, handle_cast/2]).
+
+-behaviour(gen_server).
+
+short(Url) ->
+    gen_server:call(?MODULE, {short, Url}).
+
+get(Url) ->
+    gen_server:call(?MODULE, {get, Url}).
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init(_Args) ->
+    Ets = ets:new(links, [set]),
+    {ok,Ets}.
+
+handle_call({short, LongUrl}, _From, Ets) ->
+    ShortUrl = shortening_algorithm(LongUrl),
+    EntryType = store_url(Ets, LongUrl, ShortUrl),
+    {reply, {EntryType, ShortUrl}, Ets};
+
+handle_call({get, ShortUrl}, _From, Ets) ->
+    LongUrl = search_long_url(Ets, ShortUrl),
+    {reply, LongUrl, Ets}.
+
+handle_cast(_, Ets) -> {noreply, Ets}.
+
+shortening_algorithm(Url) ->
+    Hash = crypto:hash(md4, Url),
```

Comment:
use this: base64:encode(crypto:strong_rand_bytes( N chars))
instead of this line and the following two